### PR TITLE
Fixes 20240226

### PR DESCRIPTION
**Pull request by sonjli**
+ Avoid an AV in virtual json api when response is nil
+ 64bit compliance in virtualataset
+ GetCurrent set virtual for "mockability" in in FireDAC connections
+ Fixed definition of objects field in dataset (The recursion stopped when the first object type. The second correctly was not included, but the following where not included, too)
+ Fixed conversion of floating values in json marshalling
+ Fixed a type definition in Redis KVStore

### CODE REVIEW

File: source/Json/Fido.JSON.Marshalling.pas [hunk 1]
```diff
@@ -1271,9 +1272,12 @@ class function JSONMarshaller.InternalFrom(
     tkFloat,
     tkInteger,
     tkInt64: begin
+      FormatSettings := TFormatSettings.Create;
```
Comment (mirko-bianco) [inline source/Json/Fido.JSON.Marshalling.pas]: This is tricky, because we are forcing a specific format.
We need to use the system formatting.

**Review by mirko-bianco — CHANGES_REQUESTED** [review]
As mentioned, we should use the system formatting settings or, in alternative, a way to set format settings that then will be used only by the marshallino/unmarshalling.
Like it happens with setting a specific mapper.

File: source/Json/Fido.JSON.Marshalling.pas (hunk 1 above)
Comment (sonjli) [reply source/Json/Fido.JSON.Marshalling.pas]: No, it is not the point. It is not a "internationalization" problem but exactly the opposite.
The JSON standard WANT these format settings and NOT the system settings, so it is necessary to force the JSON format. It must not be chnaged.
So the fix avoid the use of "custom" or system settings and set the JSON format settings as JSON wants.
 
immagine

Comment (wloochacz) [reply source/Json/Fido.JSON.Marshalling.pas]: > No, it is not the point. It is not a "internationalization" problem but exactly the opposite. The JSON standard WANT these format settings and NOT the system settings, so it is necessary to force the JSON format. It must not be chnaged. So the fix avoid the use of "custom" or system settings and set the JSON format settings as JSON wants.

Exactly right!

Comment (mirko-bianco) [reply source/Json/Fido.JSON.Marshalling.pas]: True. my bad.
Still this is not the place, in the code, to fix it.
I assume the type you are using is not registered in the Fido.Json.Mapping.
You can solve it by adding the proper mapping there.

**Pull request merged**